clientoptions.go

```diff
@@ -0,0 +1,150 @@
+package influxdb
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// Option is an option for the client config.
+type Option func(*Client) error
+
+// HTTPConfig is an https://github.com/influxdata/influxdb1-client compatible client for setting config
+// options.  This is here to make transition to go1 easy.
+// It is recommended that you set the options using the With___ functions instead.
+type HTTPConfig struct {
+	// Addr should be of the form "http://host:port"
+	// or "http://[ipv6-host%zone]:port".
+	Addr string
+
+	// Username is the influxdb username, optional.
+	Username string
+
+	// Password is the influxdb password, optional.
+	Password string
+
+	// UserAgent is the http User Agent, defaults to "InfluxDBClient" plus os and version info.
+	UserAgent string
+
+	// Timeout for influxdb writes, if set to zero, it defaults to a 20 second timeout. This is a difference from the influxdb1-client.
+	Timeout time.Duration
+
+	// InsecureSkipVerify gets passed to the http client, if true, it will
+	// skip https certificate verification. Defaults to false.
+	InsecureSkipVerify bool
```

Comment:
Just make this part of the TLSConfig, then you won't need to worry about their conflicting interaction.

Reply:
This (and that entire struct) is that way for compatibility with the old client.  There are better ways if you want a new client.